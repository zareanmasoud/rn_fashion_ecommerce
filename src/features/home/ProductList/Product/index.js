import Product from './Product';

export {Product as default};
