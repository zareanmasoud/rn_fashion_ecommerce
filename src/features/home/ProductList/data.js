// eslint-disable-next-line import/no-extraneous-dependencies
import R from 'res/R';

const data = {
  products: [
    {
      id: '1',
      title: 'Title',
      description: 'Description',
      discount_code: 'X9E2AD',
      photo: R.images.product1,
    },
    {
      id: '2',
      title: 'Title',
      description: 'Description',
      discount_code: 'X9E2AD',
      photo: R.images.product2,
    },
  ],
};

export default data;
