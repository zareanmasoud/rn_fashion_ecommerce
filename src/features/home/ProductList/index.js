import ProductList from './ProductList';

export {ProductList as default};
