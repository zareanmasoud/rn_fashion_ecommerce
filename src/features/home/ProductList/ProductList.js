import React, {useState, useEffect} from 'react';
import MyFlatList from 'lib/components/MyFlatList';
import Product from './Product';
import data from './data';
// import styles from './styles';

export default function ProductList() {
  const [products, setProducts] = useState([]);
  useEffect(() => {
    setProducts(data.products);
  }, []);

  return <MyFlatList data={products} itemComponent={Product} />;
}
