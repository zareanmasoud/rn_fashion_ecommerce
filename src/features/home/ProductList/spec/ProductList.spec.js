import React from 'react';
import {shallow} from 'enzyme';
import ProductList from '../ProductList';

describe('ProductList Feature', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const wrapper = shallow(<ProductList />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
