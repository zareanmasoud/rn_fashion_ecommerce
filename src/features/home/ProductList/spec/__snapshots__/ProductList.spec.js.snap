// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList Feature Rendering should match to snapshot 1`] = `
<MyFlatList
  data={Array []}
  itemComponent={[Function]}
/>
`;
